### ISSUE TITLE
Implement tiered access to VEDA Jupyterhub

### ISSUE THREAD

**Issue by batpad** [NASA-IMPACT/veda-jupyterhub#8]
Copying issue over from https://github.com/2i2c-org/infrastructure/issues/3491

From @freitagb :

## Context

We currently have two tiers of users who have been granted access to the VEDA hub, internal team members and those external. In the interest of minimizing unanticipated costs, it would be great to limit the availability of compute resources available to external users until we have a better handle on cost monitoring. As I've just learned about this functionality i'm not exactly sure how this is implemented and need guidance on how to provide better requirements. cc @j08lue @wildintellect

## Proposal

No response

## Updates and actions

No response

@yuvipanda - let's try and touch-base here and hopefully you can help sketch out at a high level what might be needed to be done here to implement this / advise if there's existing features in JupyterHub we can use to do this?

cc-ing folks on the copied over ticket:

cc @colliand @damianavila @jmunroe

**Comment by wildintellect** [comment]
For now we removed the option to spin up very large machines.
Long term we'd like 2 profiles (potentially more but 2 to start).

The first would get access to a selection of machines, typically smaller and we'd want to time cap their usage. It could be the auto-shutoff is more aggressive, it could be a limit like 8 hrs per 24 hrs (AWS Sagemaker does this), or X hours per week, etc..

The second group would get a selection of machines including larger machines, and have a different quota, or now quota at all. We probably still want an auto-shutoff time period for instances not actively used.

**Comment by batpad** [comment]
This should be straightforward to do with just configuration. If we see something like https://github.com/2i2c-org/infrastructure/blob/master/config/clusters/openscapes/common.values.yaml#L47 - we can add an `allowed_teams` list to each of the options in the ProfileList and then create different types of profiles available to different "groups" of users which map to Github Teams that have access to the hub.

Next Actions:

 - [ ] Define the different user groups / teams that we want to have
 - [ ] For each, list down what kinds of resources / profile options they should have access to
 - [ ] Make changes in https://github.com/2i2c-org/infrastructure/blob/master/config/clusters/nasa-veda/common.values.yaml#L65 to modify the profileList options and add `allowed_teams` to all the items that need to have restricted access.

@wildintellect would you be able to help or point me to the best person who would be able to help define more precisely what those Github teams should be plus what machine sizes each should have access to?

Once we have this in place, we could separately ticket time-limiting access, which may not currently be possible with configuration.

**Comment by wildintellect** [comment]
@batpad that's me and the Science Support Team https://github.com/NASA-IMPACT/veda-architecture/wiki/SC
From our initial list for testing we'll need to make clear groups on https://github.com/orgs/veda-analytics-access

What happens when someone is in more than one group?

**Comment by batpad** [comment]
> From our initial list for testing we'll need to make clear groups on https://github.com/orgs/veda-analytics-access

That sounds good - let me know if we have an initial set of groups to test with and we can do a test.

> What happens when someone is in more than one group?

I believe they would get the "maximum" of allowed permissions across both groups - i.e. access to profiles that are accessible to either group they are a part of, but we should definitely test and confirm this.

**Comment by yuvipanda** [comment]
> I believe they would get the "maximum" of allowed permissions across both groups - i.e. access to profiles that are accessible to either group they are a part of, but we should definitely test and confirm this.

correct

**Comment by batpad** [comment]
We implemented this for the SSIM-GHG workshop, where members of the workshop group were only given access to a restricted set of profiles with fixed resource requirements and base images. 

This was the PR that configured the group access to specific profile options: https://github.com/2i2c-org/infrastructure/pull/4100

Closing here. @wildintellect if there is a longer-term requirements for certain groups to have different kinds of access to resources, we can create a separate specific ticket.
